### PR TITLE
ui (histograms): Add native histogram chart to Table view

### PR DESCRIPTION
**Pull request by Maniktherana**
addresses #11269

This PR adds a better rendered view of native histograms in the table view. The histogram can be toggled between using a linear or exponential scale on the x-axis. 

https://github.com/prometheus/prometheus/assets/14011425/46056aab-2138-49ff-bfdf-c14797d8cbea

### CODE REVIEW

**Comment by Maniktherana** [comment]
@juliusv here's what I have so far. I need to add some tests and there may be room to improve how we display the total count and sum but it felt off when adding those in the same table as range + count

**Comment by Maniktherana** [comment]
Hmm, it seems the `<HistogramChart /> component isn't rendering in tests. Not sure why

**Review by juliusv — COMMENTED** [review]
Nice, I like it so far, thanks! Yeah, now seeing the count and sum outside of the bucket table, have to actually say that I prefer that too.

File: web/ui/react-app/src/themes/_shared.scss [hunk 1]
```diff
@@ -122,6 +122,128 @@ input[type='checkbox']:checked + label {
...
+  bottom: 0;
+  top: 0;
+}
+
+.histogram-bucket {
+  position: absolute;
+  width: 100%;
+  bottom: 0;
+  // background-color: #9090ff;
+  // border: 1px solid #aaaaff;
```
[the remark this answers is not in the document]
Comment (Maniktherana) [reply web/ui/react-app/src/themes/_shared.scss:232]: Not needed

File: web/ui/react-app/src/themes/_shared.scss [hunk 2]
```diff
@@ -122,6 +122,128 @@ input[type='checkbox']:checked + label {
   color: $checked-checkbox-color;
 }
 
+.summary-wrapper {
```
[the remark this answers is not in the document]
Comment (Maniktherana) [reply web/ui/react-app/src/themes/_shared.scss]: done

**Comment by Maniktherana** [comment]
@juliusv this is pretty much done I'd say, only thing is I'm having trouble understanding why the UI test is failing.

```
Test suite failed to run

    The target 'bucket-0-0-500' could not be identified in the dom, tip: check spelling

      130 |       useLocalTime: false,
      131 |     };
    > 132 |     const dataTable = mount(<DataTable {...dataTableProps} />);
          |                            ^
      133 |
      134 |     it('renders a table', () => {
      135 |       const table = dataTable.find(Table);
```

A little bit unsure how to proceed. I could just forgo testing `<HistogramChart />` in `Datatable.test.tsx` and just keep unit tests for `<HistogramChart />`

**Comment by juliusv** [comment]
@Maniktherana Strange! Yes, I think it's ok to skip testing it for now. We shouldn't be spending too much on figuring out weirdness in Enzyme tests that we will have to throw away anyway with the new Mantine-based UI and React 18.

**Comment by juliusv** [comment]
:+1: Do you want to press the "Ready for review" button to move the PR out of draft status? Then I'll merge it.

**Comment by juliusv** [comment]
Meh actually I'll just do it myself :)

**Comment by Maniktherana** [comment]
@juliusv Awesome, thanks! Ig there's some ancillary issues I can create like testing + optimizing histogram rendering

**Pull request merged**